### PR TITLE
replaced gometalinter description with golangci-lint

### PR DESCRIPTION
**Pull request by arsh-7**
Signed-off-by: Arshpreet <7arshpreet@gmail.com>

### Pull Request Checklist

* [x] I have added any new tests that need to pass to `testfile` as specified in [docs/sytest.md](https://github.com/matrix-org/dendrite/blob/master/docs/sytest.md)
* [x] Pull request includes a [sign off](https://github.com/matrix-org/dendrite/blob/master/CONTRIBUTING.md#sign-off)

### CODE REVIEW

File: CODE_STYLE.md [hunk 1]
```diff
@@ -5,17 +5,15 @@ considerations.
```
Comment (Cnly) [inline CODE_STYLE.md]: On line 3, "We follow the standard Go style using gofmt"

May be better to indicate that we actually use `goimports`, but it isn't a big problem.

File: CODE_STYLE.md [hunk 2]
```diff
@@ -5,17 +5,15 @@ considerations.
...
 
-We use `gometalinter` to run a number of linters, the exact list can be found
-in [linter.json](linter.json). Some of these are slow and expensive to run, but
-a subset can be found in [linter-fast.json](linter-fast.json) that run quickly
-enough to be run as part of an IDE.
+We use `golangci-lint` to run a number of linters, the exact list can be found under linters
+in [.golangci.yml](.golangci.yml). [Installation](https://github.com/golangci/golangci-lint#install) and [Editor Integration](https://github.com/golangci/golangci-lint#editor-integration) for it can be found in the readme of golangci-lint.
 
 For rare cases where a linter is giving a spurious warning, it can be disabled
 for that line or statement using a [comment directive](https://github.com/alecthomas/gometalinter#comment-directives), e.g.
```
Comment (Cnly) [inline CODE_STYLE.md]: The `nolint` section in the golangci-lint manual: https://github.com/golangci/golangci-lint#nolint

Comment (arsh-7) [reply CODE_STYLE.md]: whoops, fixed it now

**Comment by anoadragon453** [comment]
Merging as CI is currently known broken.

**Pull request merged**